### ISSUE TITLE
 distort_points gradcheck only verifies gradients through points — distCoeff and K paths untested

### ISSUE THREAD

**Issue by AdityaX18** [kornia/kornia#3684]
### 🚀 Feature Description

The existing `test_gradcheck` in `tests/geometry/calibration/test_distort.py` 
passes `requires_grad=True` only to `points`. The gradient paths through 
`distCoeff` and `K` have never been formally verified.

This PR adds three targeted fixes:
1. Updates `test_gradcheck` to enable `requires_grad=True` on all differentiable inputs
2. Adds `test_gradcheck_distcoeff_only` — isolates the distortion coefficient gradient path
3. Adds `test_gradcheck_K_only` — isolates the camera intrinsic matrix gradient path

No source changes. The implementation is already correctly differentiable — 
this is purely missing test coverage.

### 📂 Feature Category

Testing Infrastructure

### 💡 Motivation

The core value of a differentiable distortion model is being able to optimise 
distortion coefficients via gradient descent — NeRF camera parameter tuning, 
differentiable bundle adjustment, joint calibration during training.

The existing gradcheck only tested d(output)/d(points). It never tested:
- d(output)/d(distCoeff)  ← required for intrinsic optimisation
- d(output)/d(K)          ← required for joint K + distortion optimisation

Without these, the test suite gives false confidence that distort_points is 
fully differentiable through all its parameters. A user who attempts to 
optimise distCoeff via gradient descent has no test coverage backing that use-case.

### 💭 Proposed Solution

Modify `tests/geometry/calibration/test_distort.py`:

# BEFORE (only points has grad):
points   = torch.rand(1, 8, 2, dtype=torch.float64, requires_grad=True)
K        = torch.rand(1, 3, 3, dtype=torch.float64)                     # ← no grad
new_K    = torch.rand(1, 3, 3, dtype=torch.float64)                     # ← no grad
distCoeff = torch.rand(1, 4, dtype=torch.float64)                       # ← no grad

# AFTER (all differentiable inputs have grad):
points    = torch.rand(1, 8, 2, dtype=torch.float64, requires_grad=True)
K         = torch.rand(1, 3, 3, dtype=torch.float64, requires_grad=True)
new_K     = torch.rand(1, 3, 3, dtype=torch.float64, requires_grad=True)
distCoeff = torch.rand(1, 4, dtype=torch.float64, requires_grad=True)

Plus two new focused tests:
- test_gradcheck_distcoeff_only: fixes points/K, differentiates only through distCoeff
- test_gradcheck_K_only: fixes points/distCoeff, differentiates only through K

All 11 tests pass locally (CPU, float64, verified with pytest).

### 🔄 Alternatives Considered

Leaving the tests as-is. Rejected because it provides false confidence — 
the gradcheck passes but it is not testing the inputs that matter most 
for the differentiable calibration use-case.

### 🎯 Use Cases

1. NeRF training: optimise camera distortion coefficients jointly with the 
   scene representation via gradient descent. Requires d(loss)/d(distCoeff).

2. Differentiable bundle adjustment: minimise reprojection error end-to-end, 
   backpropagating through the distortion model into K and distCoeff.

3. Sensor fusion: train camera calibration parameters jointly with a 
   downstream model. Requires full gradient flow through all camera parameters.

### 📝 Additional Context

Verified locally:

pytest tests/geometry/calibration/test_distort.py -v -k "gradcheck"

PASSED test_gradcheck[cpu]
PASSED test_gradcheck_distcoeff_only[cpu]
PASSED test_gradcheck_K_only[cpu]
3 passed in 0.47s

Full suite: 11 passed, 0 failed.

### 🤝 Contribution Intent

- [x] I plan to submit a PR to implement this feature
- [ ] I'm requesting this feature but not planning to implement it

**Comment by Gaspard-Vernin** [comment]
Hello, I would like to contribute to this issue, can I be assigned to it ?
